test/integration/scripting_spec.js

```diff
@@ -1492,4 +1492,65 @@ describe("Interaction", () => {
       );
     });
   });
+
+  describe("in bug1782564.pdf", () => {
+    let pages;
+
+    beforeAll(async () => {
+      pages = await loadAndWait("bug1782564.pdf", getSelector("7R"));
+    });
+
+    afterAll(async () => {
+      await closePages(pages);
+    });
+
+    it("must check that charLimit is correctly set", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          await page.waitForFunction(
+            "window.PDFViewerApplication.scriptingReady === true"
+          );
+
+          await clearInput(page, getSelector("7R"));
+          // By default the charLimit is 0 which means that the input
+          // length is unlimited.
+          await page.type(getSelector("7R"), "abcdefghijklmnopq", {
+            delay: 10,
+          });
+
+          let value = await page.$eval(getSelector("7R"), el => el.value);
+          expect(value)
+            .withContext(`In ${browserName}`)
+            .toEqual("abcdefghijklmnopq");
+
+          // charLimit is set to 1
+          await page.click(getSelector("9R"));
+
+          await page.waitForFunction(
+            `document.querySelector('${getSelector(
+              "7R"
+            )}').value !== "abcdefgh"`
+          );
```

Comment:
What is this part for, since I don't fully understand why we need it (given the next check)?

Reply:
There are no guarantee that the value will have been changed "just after" having clicked on a button. So it's just a way to avoid a potential intermittent failure.